src/intrinsics.cpp

```diff
@@ -1270,6 +1275,27 @@ static jl_cgval_t emit_intrinsic(jl_codectx_t &ctx, intrinsic f, jl_value_t **ar
         ++Emitted_pointerset;
         assert(nargs == 4);
         return emit_pointerset(ctx, argv);
+
+    case add_ptr: {
+        assert(nargs == 2);
+        if (!jl_is_cpointer_type(argv[0].typ) || argv[1].typ != (jl_value_t*)jl_ulong_type)
+            return emit_runtime_call(ctx, f, argv, nargs);
+        Value *ptr = emit_unbox(ctx, ctx.types().T_ptr, argv[0], argv[0].typ);
+        Value *off = emit_unbox(ctx, ctx.types().T_size, argv[1], argv[1].typ);
+        Value *ans = ctx.builder.CreateGEP(getInt8Ty(ctx.builder.getContext()), ptr, off);
+        return mark_julia_type(ctx, ans, false, argv[0].typ);
+    }
+    case sub_ptr: {
+        assert(nargs == 2);
+        if (!jl_is_cpointer_type(argv[0].typ) || argv[1].typ != (jl_value_t*)jl_ulong_type)
```

Comment:
This is adding constraints that weren't previously there, so this is potentially breaking (and since it moved from the generic emit_untyped_intrinsic cases to here, it will also need a custom tfunc)